lib/animas/animasSimulator.js

```diff
@@ -119,13 +168,30 @@ exports.make = function(config){
       }
 
       if(currStatus !== null) {
-        if(currStatus.status === 'suspended' && event.rate === 0) {
-          //this basal is suspended
-          event.deliveryType = 'suspend';
+        if(currStatus.status === 'suspended') {
+          if (event.rate === 0) {
+            //this basal is suspended
+            event.deliveryType = 'suspend';
+          }
+          else {
+            // basal is non-zero, but status still suspended
+            // is this a suspend event we created from an alarm?
+            // If so, we should submit a joined suspend/resume event
```

Comment:
I think I might be a little confused by this comment and all the branches flying around. It's not just these deviceEvent suspend/resumes that will need to be combined into a single event for upload to the dataservices API...so shouldn't it be done in common somewhere? Maybe you're already reworking this somewhere else?